homeassistant/components/stream/__init__.py

```diff
@@ -52,14 +53,18 @@ def request_stream(hass, stream_source, *, fmt='hls',
     # For RTSP streams, prefer TCP
     if isinstance(stream_source, str) \
             and stream_source[:7] == 'rtsp://' and not options:
-        options['rtsp_flags'] = 'prefer_tcp'
+        if stream_protocol == 'udp':
+        	options['rtsp_flags'] = 'prefer_upd'
```

Comment:
probably prefer_udp

Reply:
Have you tried this patch ?

From the documentation (https://ffmpeg.org/ffmpeg-protocols.html) this option doesn't exist...

Reply:
Stream integration does not use ffmpeg

Reply:
got it, but https://libav.org/documentation/avconv.html doesn't even have 'prefer_tcp'

Reply:
Ah, you're right.

Reply:
Yeah, that is an invalid option. Also, the stream component does use python bindings for FFMPEG’s underlying C libraries. It does not use the FFMPEG CLI, but the options are the same.

Reply:
Then again, `prefer_tcp` is supposed to just attempt a TCP connection first and presumably fall back to UDP, which it looks like it doesn’t do.